dask/array/tests/test_array_core.py

```diff
@@ -1212,14 +1212,11 @@ def test_reshape_splat():
     assert_eq(x.reshape((25,)), x.reshape(25))
 
 
-def test_reshape_fails_for_dask_only():
-    cases = [((3, 4), (4, 3), 2)]
-    for original_shape, new_shape, chunks in cases:
-        x = np.random.randint(10, size=original_shape)
-        a = from_array(x, chunks=chunks)
-        assert x.reshape(new_shape).shape == new_shape
-        with pytest.raises(ValueError):
-            da.reshape(a, new_shape)
+def test_reshape_not_implemented_error():
+    a = da.ones((4, 5, 6), chunks=(2, 2, 3))
+    for new_shape in [(2, 10, 6), (5, 4, 6), (6, 5, 4)]:
+        with pytest.raises(NotImplementedError):
```

Comment:
There's only one place in reshape that raises a `NotImplementedError`, and we don't actually care to test what the contents of the error message say (it's a fixed message, not one generated by any user-input data). I don't see a reason to make more adjustments to such a small PR.

Reply:
IMO it's best practice to make sure you're testing the right code path. For example, new errors can be added, it can raise an error elsewhere, etc. YMMV 🤷‍♂️